.github/workflows/java-roll-version.yaml

```diff
@@ -0,0 +1,39 @@
+# A workflow to find the current version from pom and then commit the release version (with tag) and the next dev cycle version
+# Note this uses java-java-update-version.yaml
+name: Roll maven version
+on:
+  workflow_call:
+
+jobs:
+  extract:
+    name: extract-version
+    runs-on:
+      - self-hosted
+      - java
+    outputs:
+      major: ${{ steps.major.outputs.major }}
+      minor: ${{ steps.minor.outputs.minor }}
+      patch: ${{ steps.patch.outputs.patch }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: major
+        run: |
+          echo ::set-output name=major::$(mvn build-helper:parse-version help:evaluate -Dexpression=parsedVersion.majorVersion -q -DforceStdout --no-transfer-progress)
```

Comment:
You may want to change these; the `::set-output` syntax is deprecated and slated to be removed.

Use `$GITHUB_OUTPUT` instead:

```bash
major=$(
  mvn --quiet build-helper:parse-version \
    help:evaluate
    -Dexpression=parsedVersion.majorVersion \
    -DforceStdout \
    --no-transfer-progress
)
echo "major=$major" >> "$GITHUB_OUTPUT"
```